lib/ProvidePlugin.js

```diff
@@ -31,10 +31,14 @@ ProvidePlugin.prototype.apply = function(compiler) {
 				parser.plugin("expression " + name, function(expr) {
 					var nameIdentifier = name;
 					var scopedName = name.indexOf(".") >= 0;
+					var expression = "require(" + JSON.stringify(request[0]) + ")";
 					if(scopedName) {
 						nameIdentifier = "__webpack_provided_" + name.replace(/\./g, "_dot_");
 					}
-					if(!ModuleParserHelpers.addParsedVariable(this, nameIdentifier, "require(" + JSON.stringify(request) + ")")) {
+					if(request.length > 1) {
+						expression += '.' + request.slice(1).join('.');
```

Comment:
Try to use something like this:

`expression += request.slice(1).map(function(r) { return "[" + JSON.stringify(r) + "]"; }.join("");`

This is safe to non-identifier chars in the properties.